edx/edx-enterprise
# ENT-210: Remove unnecessary pipeline element; allow ECU creation when consent is disabled

**Description:** Previously, users would not be linked as EnterpriseCustomerUsers when data sharing consent was disabled. This PR remedies that, and also removes a third-party auth pipeline element that is no longer necessary due to previous changes.

**Sandbox URL:** https://pr14567.sandbox.opencraft.hosting (admin user: staff/edx)

**JIRA:** [ENT-210](https://openedx.atlassian.net/browse/ENT-210)

**Dependencies:** Requires edx/edx-platform#14567

**Merge deadline:** 13 March 2017

**Installation instructions:**

1. Set up a devstack with the branch in the linked edx-platform PR.
2. Set up a TestShib SAML integration with the instructions [found here](http://edx.readthedocs.io/projects/edx-installing-configuring-and-running/en/latest/configuration/tpa/tpa_SAML_SP.html).
3. Create an EnterpriseCustomer in the Django admin.

**Testing instructions:**

1. Link the EnterpriseCustomer to the TestShib SAML integration
2. Verify that data sharing consent is set to disabled on the EnterpriseCustomer
3. In an incognito window, browse to the LMS registration page
4. Attempt to register via TestShib SAML
5. After registration, observe that you are logged in directly.
6. In the Django Admin panel for the EnterpriseCustomer, click Manage Learners
7. Observe that your SAML-created user is linked as an Enterprise Learner.

@pomegranited for initial review

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**
